### ISSUE TITLE
Add `linear` as a strategy to `fill_null` 

### ISSUE THREAD

**Issue by tim-habitat** [pola-rs/polars#8929]
### Problem description

It would be cool to have a linear interpolation strategy to fill missing values. 
In particular, this could be super useful after an upsample. As in, upsample then interpolate the nulls between the previous values.

**Comment by ritchie46** [comment]
You can already do this.

```
df.upsample().select(pl.all().interpolate())
```

**Comment by kolergy** [comment]
The linear interpolation strategy would be a more coherent way to do it from the point of view of the user.

**Comment by ritchie46** [comment]
No, it would be yet another way to do it. I am opposed to pushing compute into keyword arguments if there is a clear way to do that. If you want to `interpolate`, you must use the `interpolate` expression.
